src/features/defi/providers/univ2/components/UniV2Manager/Deposit/components/Approve.tsx

```diff
@@ -29,40 +34,56 @@ import {
 } from 'state/slices/selectors'
 import { useAppSelector } from 'state/store'
 
-import { FoxEthLpDepositActionType } from '../DepositCommon'
+import { UniV2DepositActionType } from '../DepositCommon'
 import { DepositContext } from '../DepositContext'
 
-type FoxEthLpApproveProps = StepComponentProps & {
+type UniV2ApproveProps = StepComponentProps & {
   accountId: AccountId | undefined
   onNext: (arg: DefiStep) => void
 }
 
-const moduleLogger = logger.child({ namespace: ['FoxEthLpDeposit:Approve'] })
+const moduleLogger = logger.child({ namespace: ['UniV2Deposit:Approve'] })
 
-export const Approve: React.FC<FoxEthLpApproveProps> = ({ accountId, onNext }) => {
+export const Approve: React.FC<UniV2ApproveProps> = ({ accountId, onNext }) => {
   const { state, dispatch } = useContext(DepositContext)
   const estimatedGasCryptoPrecision = state?.approve.estimatedGasCryptoPrecision
   const translate = useTranslate()
   const mixpanel = getMixPanel()
+  const { query } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, assetNamespace, assetReference } = query
   const { lpAccountId } = useFoxEth()
-  const { approve, allowance, getDepositGasDataCryptoBaseUnit } =
-    useFoxEthLiquidityPool(lpAccountId)
 
-  const foxEthLpOpportunityFilter = useMemo(
+  const lpAssetId = toAssetId({ chainId, assetNamespace, assetReference })
+
+  const lpOpportunityFilter = useMemo(
     () => ({
-      lpId: foxEthLpAssetId,
-      assetId: foxEthLpAssetId,
+      lpId: lpAssetId as LpId,
+      assetId: lpAssetId,
       accountId,
     }),
-    [accountId],
+    [accountId, lpAssetId],
+  )
+  const lpOpportunity = useAppSelector(state =>
+    selectEarnUserLpOpportunity(state, lpOpportunityFilter),
   )
-  const foxEthLpOpportunity = useAppSelector(state =>
-    selectEarnUserLpOpportunity(state, foxEthLpOpportunityFilter),
+
+  const assetId0 = lpOpportunity?.underlyingAssetIds[0] ?? ''
+  const assetId1 = lpOpportunity?.underlyingAssetIds[1] ?? ''
+  const asset1ContractAddress = useMemo(
+    () => ethers.utils.getAddress(fromAssetId(assetId1).assetReference),
+    [assetId1],
   )
+  const { approve, allowance, getDepositGasDataCryptoBaseUnit } = useUniV2LiquidityPool({
+    accountId: lpAccountId ?? '',
+    lpAssetId,
+    assetId0,
+    assetId1,
+  })
+
   const assets = useAppSelector(selectAssets)
-  const foxAsset = useAppSelector(state => selectAssetById(state, foxAssetId))
+  const asset1 = useAppSelector(state => selectAssetById(state, assetId1))
   const feeAsset = useAppSelector(state => selectAssetById(state, ethAssetId))
-  if (!foxAsset) throw new Error('Missing FOX asset')
+  if (!asset1) throw new Error('Missing asset 1')
```

Comment:
might want a check for asset 0 here - eth/fox specifically had eth as the fee asset but that won't be the case with most LP pairs

Reply:
b53ac4f1cc8e82a3806cd91d4f8a97fbd4792c90